openedx/core/djangoapps/user_authn/api/required_fields.py

```diff
@@ -0,0 +1,152 @@
+"""
+Optional Fields API used by Authn MFE to populate progressive profiling form
+"""
+import copy
+
+from django.conf import settings
+from edx_rest_framework_extensions.auth.jwt.authentication import JwtAuthentication
+from rest_framework import status
+from rest_framework.authentication import SessionAuthentication
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+from rest_framework.throttling import UserRateThrottle
+from rest_framework.views import APIView
+
+from openedx.core.djangoapps.site_configuration import helpers as configuration_helpers
+from openedx.core.djangoapps.user_authn.api import form_fields
+from openedx.core.lib.api.authentication import BearerAuthentication
+
+
+class RequiredFieldsView(APIView):
```

Comment:
Lets create a class called "RegistrationFieldsView` which contains these common variables that are used by both optional and required fields view. i.e `EXTRA_FIELDS` and `def _get_field_order`